django/forms/models.py

```diff
@@ -1317,8 +1317,11 @@ def _check_values(self, value):
                     params={'pk': pk},
                 )
         qs = self.queryset.filter(**{'%s__in' % key: value})
-        pks = {str(getattr(o, key)) for o in qs}
-        for val in value:
+        return self.validate_choices(qs, key, value)
+
+    def validate_choices(self, qs, field_name, values):
```

Comment:
I think `queryset` would be a more natural name.

Reply:
Can you elaborate on what you mean here also?

Reply:
I think he means that the arg should be "queryset" instead of "qs".